src/clarity-angular/tabs/vtabs.ts

```diff
@@ -0,0 +1,212 @@
+     * 4. Sets the first link and first content as active, if needed.
+     *
+     * No params, called as part of the angular lifecycle
+     */
+    ngAfterContentInit() {
+        // Initialize tabs and content.
+
+        this.vtabLinks.forEach((link, index) => {
+            link.id = "clr-vtab-link-" + nbVTabLinks.toString();
+            nbVTabLinks++;
+            this.mapContentId(index, link.id);
+            // subscribe to the links click event
+            link.activatedChange.subscribe((updatedLink: VTabsNavLink) => {
+                this.updateForActiveLink(updatedLink);
+            });
+
+            //check for activated link
+            if (link.activated) {
+                this.activatedLink = index;
+            }
+        });
+
+        // Activate the first tab and content if nothing has been set with the VTabsNavLink.activated @Input().
+        if (!this.activatedLink) {
+            let testLink = this.vtabLinks.first;
+            testLink.activated = true;
+
+            let testContent: VTabsContent = this.vtabContents.first;
+            testContent.activated = true;
+        } else {
+            this.activateTabContentAtIndex(this.activatedLink);
+        }
+    }
+
+    /**
+     * @function updateForActiveLink
+     *
+     * @description
+     * Method to update all the VTabNavLinks when an activatedChange event is heard by:
+     *
+     * 1. Deactivating all the linkItem.ids that don't match the updatedLink passed as arg
+     * 2. Identifying the index of the newly activated tab link so we can activate corresponding VTabContent
+     * 3. Activating the corresponding VTabsContent item.
+     *
+     * @param updatedLink -  a VTabNavLink that was clicked on.
+     *
+     */
+    updateForActiveLink(updatedLink: VTabsNavLink) {
+        let linkIndex: number; // Map between link and content
+        this.vtabLinks.forEach((linkItem: VTabsNavLink, index: number) => {
+            if (linkItem.id !== updatedLink.id) {
+                linkItem.activated = false;
+            } else {
+                linkIndex = index; // Position of the link that was clicked.
+                this.activatedLink = index;
+            }
+        });
+
+        // Update content for index
+        if (typeof linkIndex !== "undefined") { // Remember, remember 0 is also false.
+            this.activateTabContentAtIndex(linkIndex);
+        }
+
+        // Broadcast the link change event
+        this.linkChange.emit(updatedLink);
+    }
+
+    /**
+     * @function activateTabContentAtIndex
+     *
+     * @description
+     * Method to update VTabContent items as follows:
+     *
+     * 1. Deactivate all VTabContent items not equal to the linkIndex arg
+     * 2. Activate the VTabContent iotem that has the same index as the linkIndex arg
+     *
+     * @param linkIndex - the index to activate in this.vtabsContent QueryList
+     */
+    activateTabContentAtIndex(linkIndex: number) {
```

Comment:
Did we write any tests for this? I see very high-level tests. But not many tests for the API in isolation.